front/src/utils/date.ts

```diff
@@ -47,6 +47,16 @@ export const dateTimeToIso = (inputDateTime: string) => {
   return null;
 };
 
+export const dateTimeZToIso = (inputDateTime: string) => {
+  // Regex to check format 1234-56-78T12:00:00Z
+  const inputDateTimeRegex = /^\d{4}-\d{2}-\d{2}[T\s]\d{2}:\d{2}(?::\d{2}){0,1}(Z)?$/;
+  if (inputDateTimeRegex.test(inputDateTime)) {
+    const userTimeZone = dayjs.tz.guess(); // Format : 'Europe/Paris'
+    return dayjs.utc(inputDateTime).tz(userTimeZone).format();
+  }
+  return '';
+};
```

Comment:
Should I modify the existing function `dateTimeToIso`, or create a new one as I did? Since my argument can't be null, I need to return a string or ''